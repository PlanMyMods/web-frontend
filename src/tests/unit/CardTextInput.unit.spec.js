import { shallowMount } from "@vue/test-utils";
import CardTextInput from "@/components/CardTextInput.vue";

describe("CardTextInput", () => {
  const wrapper = shallowMount(CardTextInput, {
    propsData: {
      label: "Name",
    },
  });
  test("should match snapshot", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
  test("should render the correct label", () => {
    expect(wrapper.find(".text-sm").text()).toBe("Name");
  });
  test("should update the value when input event is emitted", async () => {
    const input = wrapper.find("input");
    input.setValue("John Doe");
    input.trigger("input");
    await wrapper.vm.$nextTick();
    expect(wrapper.vm.value).toBe("John Doe");
  });
});
