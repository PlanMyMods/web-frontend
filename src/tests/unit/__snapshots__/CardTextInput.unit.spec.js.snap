// Vitest Snapshot v1

exports[`CardTextInput > should match snapshot 1`] = `
"<div class=\\"flex flex-col flex-nowrap box-border\\">
  <div class=\\"whitespace-normal text-sm font-medium text-gray-700 dark:text-gray-300\\">Name</div>
  <div class=\\"bg-white dark:bg-gray-800 border dark:border-gray-600 drop-shadow mt-3 mb-1\\"><input class=\\"relative inline-flex flex-auto whitespace-nowrap focus:outline-none text-left text-base select-auto w-full z-10 p-2 text-gray-700 dark:text-gray-300 dark:bg-gray-800\\" name=\\"Name\\"></div>
</div>"
`;
